Calculate/Program.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Linq.Expressions;
+using System.Runtime.CompilerServices;
+using Microsoft.VisualBasic;
+
+namespace Calculate;
+
+public class Program
+{
+    public Func<string?> ReadLine { get; init; } = () => { return Console.ReadLine(); };
+    public Action<string> WriteLine { get; init; } = (value) => { Console.WriteLine(value); };
+
+    public Program()
+    {
+    }
+
+    public Program(Func<string> readLine, Action<string> writeLine)
+    {
+        ReadLine = readLine;
+        WriteLine = writeLine;
+    }
+
+    static void Main()
+    {
+        Program program = new();
+        Calculator calculator = new();
+
+        program.WriteLine("Enter a calculation (e.g.,3 + 4), or blank to quit");
+
+        string? input;
+        while (!string.IsNullOrWhiteSpace(input = program.ReadLine()))
```

Comment:
Consider that having an infinite loop creates a certain kind of interactive experience like a [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop) vs if you did not have the loop the app would exit on each invocation and could be used like in an automated tool call fashion.